src/main/java/bowling/domain/Bowling.java

```diff
@@ -38,4 +40,18 @@ public int nextFrame() {
         .collect(Collectors.toList());
   }
 
+
+  public List<Score> scores() {
+    List<Score> scores = new ArrayList<>();
+
+    Score score = null;
+    for (Frame frame : frames) {
+      score = frame.score(score);
+      scores.add(score);
+    }
+
+    return scores.stream()
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
```

Comment:
중간의 `frame.score(score)` 부분이 조금 어려워 보이기는 하지만,
`.peek()` 과 같은 중간 처리 연산자를 응용하신다면, 별도의 for loop과 `List` 객체 할당 없이
처음부터 끝까지 하나의 스트림 체인으로 구현하실 수 있으리라 생각됩니다.
한번 시도해보시겠어요?

Reply:
쉽게 생각을 했는데 답이 안 없네요.
계속 고민을 한 번 해 보겠습니다.
(저도 하나로 만들고 싶어서 PR 보내기 전에 이렇게 저렇게 해 본 곳이라서..)

Reply:
```suggestion
    frames.stream()
        .reduce((Score) null, (s, f) -> {
          Score score = f.score(s);
          if (score != null)
            scores.add(score);
          return score;
        }, (a, b) -> b);
    return scores;
```

이렇게는 구현 할 수 있는데 개인적으로 깔끔하지 않네요.